ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/TestXAResourceRecovery.java

```diff
@@ -30,21 +30,32 @@
 public class TestXAResourceRecovery implements XAResourceRecovery {
 
 	private static Stack<XAResource> resources = new Stack<XAResource>();
+	private static int count = 0;
 
 	public XAResource getXAResource() throws SQLException {
-		return resources.pop();
+		count--;
+		XAResource toReturn = resources.remove(0);
+		resources.push(toReturn);
```

Comment:
OK so you removed the item from the bottom of the stack and put it on the top of the stack. I don't know why you are doing that so some comments are in order.